### ISSUE TITLE
PDAL 2.3.0: pdal_utils_test fails on i686/32bit architecture

### ISSUE THREAD

**Issue by neteler** [PDAL/PDAL#3469]
**Describe the bug**

I am one of the PDAL packagers on Fedora and I have tried to package v2.3.0 (i.e., updated the SPEC file and patches as needed. Unfortunately there seems to be a bug with the 32bit build:

https://kojipkgs.fedoraproject.org//work/tasks/9253/69009253/build.log

```
...
 24/116 Test  #24: pdal_support_test ......................   Passed    0.10 sec
        Start  25: pdal_utils_test
 25/116 Test  #25: pdal_utils_test ........................Subprocess aborted***Exception:   0.28 sec
[==========] Running 20 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 20 tests from UtilsTest
[ RUN      ] UtilsTest.test_random
[       OK ] UtilsTest.test_random (0 ms)
[ RUN      ] UtilsTest.test_comparators
[       OK ] UtilsTest.test_comparators (0 ms)
[ RUN      ] UtilsTest.test_base64
[       OK ] UtilsTest.test_base64 (0 ms)
[ RUN      ] UtilsTest.blanks
[       OK ] UtilsTest.blanks (0 ms)
[ RUN      ] UtilsTest.split
[       OK ] UtilsTest.split (0 ms)
[ RUN      ] UtilsTest.splitChar
[       OK ] UtilsTest.splitChar (0 ms)
[ RUN      ] UtilsTest.split2
[       OK ] UtilsTest.split2 (0 ms)
[ RUN      ] UtilsTest.split2Char
[       OK ] UtilsTest.split2Char (0 ms)
[ RUN      ] UtilsTest.case
[       OK ] UtilsTest.case (0 ms)
[ RUN      ] UtilsTest.starts
[       OK ] UtilsTest.starts (0 ms)
[ RUN      ] UtilsTest.iequals
[       OK ] UtilsTest.iequals (0 ms)
[ RUN      ] UtilsTest.replaceAll
[       OK ] UtilsTest.replaceAll (0 ms)
[ RUN      ] UtilsTest.escapeNonprinting
[       OK ] UtilsTest.escapeNonprinting (0 ms)
[ RUN      ] UtilsTest.wordWrap
[       OK ] UtilsTest.wordWrap (0 ms)
[ RUN      ] UtilsTest.wordWrap2
[       OK ] UtilsTest.wordWrap2 (0 ms)
[ RUN      ] UtilsTest.simpleWordexpTest
[       OK ] UtilsTest.simpleWordexpTest (0 ms)
[ RUN      ] UtilsTest.naninf
[       OK ] UtilsTest.naninf (0 ms)
[ RUN      ] UtilsTest.numeric_cast
[       OK ] UtilsTest.numeric_cast (0 ms)
[ RUN      ] UtilsTest.escapeJSON
[       OK ] UtilsTest.escapeJSON (0 ms)
[ RUN      ] UtilsTest.map
pdal_utils_test: /builddir/build/BUILD/PDAL-2.3.0-src/test/unit/UtilsTest.cpp:519: virtual void pdal::UtilsTest_map_Test::TestBody(): Assertion `ctx.addr()' failed.
```

**Expected behavior**

The test should not fail on 32bit platforms.

**System/installation information:**

Fedora 35 build system:
- Architecture: i686
- details in https://kojipkgs.fedoraproject.org//work/tasks/9253/69009253/hw_info.log
- Bug report: https://bugzilla.redhat.com/show_bug.cgi?id=1965634

Log files:
- https://kojipkgs.fedoraproject.org//work/tasks/9253/69009253/

**Comment by abellgithub** [comment]
The test just shouldn't run on a 32-bit system, since it's explicitly attempting to map a file that won't fit in a 32-bit address space.

**Comment by neteler** [comment]
Thanks, I see. So I'll blacklist it somehow.

**Comment by abellgithub** [comment]
I'm fixing it. I'm just letting you know.

**Comment by abellgithub** [comment]
Fixed by #3489.
This should perhaps be changed to use sizeof(size_t) instead.
